algorithms_task_2/student/include/impl.h

```diff
@@ -9,4 +9,12 @@
  *
  * @return true if source starts from a prefix
  */
-bool startsWith(const std::string& source, const std::string& prefix);
+bool startsWith(const std::string& source, const std::string& prefix)
+{
+    if (prefix.size() > source.size()) return false;
+    return prefix.cend() == std::mismatch(prefix.cbegin(), prefix.cend(), source.cbegin()).first;
+
+    //alternative
+//    if (prefix.size() > source.size()) return false;
+//    return std::equal(prefix.cbegin(), prefix.cend(), source.cbegin());
```

Comment:
This is better alternative for this case, because it does not require checking the iterator